domain/storage/service/storagepool.go

```diff
@@ -63,15 +64,18 @@ func (s *StoragePoolService) validateConfig(name string, providerType storage.Pr
 	if providerType == "" {
 		return storageerrors.MissingPoolTypeError
 	}
-	if s.registry == nil {
-		return errors.Errorf("cannot validate storage provider config for %q without a registry", name)
-	}
 
 	cfg, err := storage.NewConfig(name, providerType, attrs)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	p, err := s.registry.StorageProvider(providerType)
+
+	registry, err := s.registryGetter.GetStorageRegistry(ctx)
```

Comment:
Since the registryGetter returns a cached copy, it's cheap to keep calling GetStorageRegistry every time we validate. I wonder if we want to add a comment to that effect. Any time you see a ctx being passed in, you wonder if the call might potentially be expensive to make. You'd not want validation to be gated on an expensive api call every time.
Same comment in the other place this is used.

Reply:
Sure can :-1: